packages/components/src/popover/index.js

```diff
@@ -62,48 +62,135 @@ function useThrottledWindowScrollOrResize( handler, ignoredScrollableRef ) {
+	anchorRefFallback,
+	anchorRect,
+	getAnchorRect,
+	anchorRef = false,
+	anchorIncludePadding
+) {
+	if ( anchorRect ) {
+		return anchorRect;
+	}
+
+	if ( getAnchorRect ) {
+		if ( ! anchorRefFallback.current ) {
+			return;
+		}
+
+		return getAnchorRect( anchorRefFallback.current );
+	}
+
+	if ( anchorRef !== false ) {
+		if ( ! anchorRef ) {
+			return;
+		}
+
+		if ( anchorRef instanceof window.Range ) {
+			return getRectangleFromRange( anchorRef );
+		}
+
+		const rect = anchorRef.getBoundingClientRect();
+
+		if ( anchorIncludePadding ) {
+			return rect;
+		}
+
+		return withoutPadding( rect, anchorRef );
+	}
+
+	if ( ! anchorRefFallback.current ) {
+		return;
+	}
+
+	const { parentNode } = anchorRefFallback.current;
+	const rect = parentNode.getBoundingClientRect();
+
+	if ( anchorIncludePadding ) {
+		return rect;
+	}
+
+	return withoutPadding( rect, parentNode );
+}
+
+function addBuffer( rect, verticalBuffer = 0, horizontalBuffer = 0 ) {
+	return {
+		x: rect.left - horizontalBuffer,
+		y: rect.top - verticalBuffer,
+		width: rect.width + ( 2 * horizontalBuffer ),
+		height: rect.height + ( 2 * verticalBuffer ),
+		left: rect.left - horizontalBuffer,
+		right: rect.right + horizontalBuffer,
+		top: rect.top - verticalBuffer,
+		bottom: rect.bottom + verticalBuffer,
+	};
+}
+
+function withoutPadding( rect, element ) {
+	const {
+		paddingTop,
+		paddingBottom,
+		paddingLeft,
+		paddingRight,
+	} = window.getComputedStyle( element );
+	const top = paddingTop ? parseInt( paddingTop, 10 ) : 0;
+	const bottom = paddingBottom ? parseInt( paddingBottom, 10 ) : 0;
+	const left = paddingLeft ? parseInt( paddingLeft, 10 ) : 0;
+	const right = paddingRight ? parseInt( paddingRight, 10 ) : 0;
+
+	return {
+		x: rect.left + left,
+		y: rect.top + top,
+		width: rect.width - left - right,
+		height: rect.height - top - bottom,
+		left: rect.left + left,
+		right: rect.right - right,
+		top: rect.top + top,
+		bottom: rect.bottom - bottom,
+	};
+}
+
 /**
  * Hook used to compute and update the anchor position properly.
  *
- * @param {Object} anchorRef       reference to the popover anchor element.
- * @param {Object} contentRef      reference to the popover content element.
- * @param {Object} anchorRect      anchor Rect prop used to override the computed value.
- * @param {Function} getAnchorRect function used to override the anchor value computation algorithm.
+ * @param {Object}   anchorRefFallback      Reference to the popover anchor fallback element.
+ * @param {Object}   contentRef             Reference to the popover content element.
+ * @param {Object}   anchorRect             Anchor Rect prop used to override the computed value.
+ * @param {Function} getAnchorRect          Function used to override the anchor value computation algorithm.
+ * @param {Object}   anchorRef              Reference to the popover anchor fallback element.
+ * @param {Object}   anchorIncludePadding   Whether to include the anchor padding.
```

Comment:
Type should be boolean.

Reply:
The buffer props below should have their type corrected too.

Similarly, `anchorRef` should be better defined, as it seems to assume many shapes: `Range`, `Boolean`, maybe also `WPElement`? Do we need a new type definition for it, and is the "ref" portion of the name still apt?

Finally, I recommend that `anchorIncludePadding` be rephrased so as to convey that it's a Boolean, with a `should` prefix.

Reply:
I updated the types. `anchorRef` can be an `Element` or a `Range`. I think the naming is fine as there are references to a live element or range.

I changed `anchorIncludePadding` to `shouldAnchorIncludePadding`, but I find the naming a bit strange. Maybe `includeAnchorPadding` or `shouldIncludeAnchorPadding`?

Reply:
I think all names will be strange :P. This is when I miss Scheme's or Ruby's use of question marks for predicates (e.g. `(empty? x)` or `x.empty?`) — not that this helps in the conversation.

There's also `anchorShouldIncludePadding`? Honestly, as long as `should` or a comparable modal is there, I'm happy with whatever. :)